src/userbase-js/src/db.js

```diff
@@ -128,6 +129,14 @@ class Database {
         }
         this.unverifiedTransactions[j].addTransaction(transactions[i], transactionCode)
       }
+
+      if (this.init && transactionCode === 'Success') {
+        this.onChange(this.getItems())
+      }
+    }
+
+    if (!this.init) {
+      this.onChange(this.getItems())
     }
```

Comment:
What's the use of this.init here? And why do we call onChange from 2 places?

Reply:
This is an optimization that should be just as safe as only calling it in 1 place inside the for loop.

The `transactions` array passed into this `applyTransactions` function will only have >1 transactions inside it in 2 circumstances:
1. When opening a database the first time (could be a high number of transactions)
2. When the user attempts concurrent transactions (will usually be a pretty low number of transactions)

Imagine a user opens their database for the first time and they have lots of transactions in their transaction log. Without this optimization, the changeHandler would fire for every single transaction in their log.

When opening the database for the first time (i.e. `this.init` is false), we assume the client is unable to apply any other transactions aside from the first batch it receives. Considering this is our assumption inherent in the structure of the code, it's safe to wait until all transactions in the first batch are applied before firing the change handler (working on guaranteeing this assumption right now with the queue implementation we discussed).

I get that something like this might be considered pre-mature optimization, but it also allows all the tests I already have written that expect the change handler to be called when opening the database to pass. So it's an optimization that also saves me the extra work of rewriting tests :)

I was going to comment an explanation for this in the code but found it pretty complicated when typing up.

Reply:
If this explanation^ is satisfactory, I can write a comment in the code that has a hyperlink to it :)